python/src/aica_api/client.py

```diff
@@ -47,11 +47,16 @@ def _ws_endpoint(self, endpoint=''):
         """
         return f'{self._ws_address}/{endpoint}'
 
-    def check(self) -> requests.Response:
+    def check(self) -> bool:
         """
-        Make a GET request to the default endpoint to verify connectivity.
+        Check if the API version is v2 (any v2.x.x tag)
         """
-        return requests.get(self._endpoint())
+        # TODO: come up with a compatibility table in the future
+        api_version = requests.get(f'{self._address}/version').json()
```

Comment:
suggestion: I would also add a try / catch around this line, that way we can also check for connection errors (i.e. API server is not running or is on a different port / address)

```python
try:
    api_version = requests.get(f'{self._address}/version').json()
except requests.exceptions.RequestException as e:
    print(f'Error connecting to the API! {e}')
    return False
```